File: arrays/reverseAnArray.cpp
#include<iostream>
#include<vector>
using namespace std;

void reverseArray(vector<int> arr) {
    int left = 0;
    int right = arr.size()  - 1;
    int temp = 0;

    while(left < right) {
        temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
        left++;
        right--;
    }
    for(int i = 0;i<arr.size();i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    vector<int> arr = {10,20,30}; //reverse -> 30,20,10

    reverseArray(arr);
}

File: arrays/reverse_an_array.py
#write a function to reverse an array

def reverse_array(arr):
    low = 0
    high = len(arr) - 1
    
    while(low < high):
        arr[low] , arr[high] = arr[high] , arr[low]
        low +=1 
        high -=1
        
    return arr


array = [10,30,23,22,101,2,3,4,5,12]
print(reverse_array(array)) # [12, 5, 4, 3, 2, 101, 22, 23, 30, 10]